### PR TITLE
fix(jobsdbFooter): fix footer padding issue on tablet

### PR DESCRIPTION
**Pull request by hscgavin**
** Please provide as much detail as possible, but feel free to remove irrelevant sections **
2 issues:
image
image

also need some change in srp

```
.expandedFooter{
  padding-bottom: @grid-base * 5;
  @media @desktop {
   padding-bottom: 0;
  }
}
```

after:
oct-08-2018 16-03-51

### CODE REVIEW

**Comment by mengtzu** [comment]
This is a good change (moving from `@mobile` should be done everywhere, as the breakpoint change has shown it's brittle and bad practice unless we really need it).

I think though the reason for this should be documented, and maybe we should expose a prop to control it.  The problem is created by the header, not the footer.  If you used the footer for some reason *without* using the header, this would be weird.  That should be a rare case, but I think accomodating it (with a default false prop) would be appropriate.

Also shouldn't JobStreet need this as well?

**Comment by hscgavin** [comment]
@mengtzu I don't think Jobstreet has `expandedFooter`. I think can fix the pagination when we are working on jobstreet. the root cause is we added paddings on different elements, on mobile we add padding to children in order to pushup the footer. on desktop we add padding to wrapper.. .. It's all about consistency. I am not sure what can be documented? Please advise 🙇

**Comment by patrickliu-sa** [comment]
is it possible to hoist some global scoped prop that can be cross-shared for all components?  🤔

**Comment by mengtzu** [comment]
@hscgavin it's important people understand the reasoning for the padding.  It's not a requirement of the footer, it's not even the footer's business, the footer just happens to be at the spot in the DOM that needs more space.

The padding is there because of the *header*.  If you weren't using the header, you would not need this padding, and in fact you'd have a useless bit of empty space hanging around for no reason.  That is why I propose a prop to control it, and that's what you can document - the *reason* for this tech debt.

This is also why I expect JobStreet will need the change (assuming it wasn't implemented mobile-first from the start) - regardless of the characteristics of the footer, if you're using the header you need this space.  That should be equally true for JobStreet.

**Comment by hscgavin** [comment]
@mengtzu @patrickliu-sa , a prop, to detect the device? isMobile, isTablet, isDesktop? 🤔Please elaborate. 
as far as I know, we use Mobile first css (`@desktop`) to hide top header and show bottom header, in footer we use desktop first css `@mobile` to add padding, after we changed the desktop breakpoint (although mobile break point stay the same), we got the broken ux.

**Comment by mengtzu** [comment]
@hscgavin you're correct, there's no need to have a prop to detect device.

I'm proposing a prop so that you can use the footer without using the header.

**Comment by patrickliu-sa** [comment]
We need a reference for CSS in global scope that can be set in component level.  Can we make use of custom properties on CSS space, or even data attributes can help (`<body data-device-type="mobile">`)?

File: jobsDB/Footer/footer.less [hunk 1]
```diff
@@ -38,7 +45,7 @@
 .pullRight {
   padding: 5px 0;
   @media @desktop {
-    padding: 12px 0;
+    padding: (@grid-base * 3) 0;
```
Comment (ashin) [inline jobsDB/Footer/footer.less]: nice clean up 👍

File: jobsDB/Footer/Footer.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 import styles from './footer.less';
 import PropTypes from 'prop-types';
+import classnames from 'classnames';
 
 import FooterLinks from './components/FooterLinks/FooterLinks';
 import FooterHK from './FooterHK/FooterHK';
 import localization from './localization';
 
-const Footer = ({ language, country, isExpandedVersion, ...restProps }) => {
+const Footer = ({ language, country, isExpandedVersion, showHeaderActionTrayOffset = true, ...restProps }) => {
```
Comment (ashin) [inline jobsDB/Footer/Footer.js]: As ugly as that prop name is, I like that it's explicitly expressed in code.

Comment (mengtzu) [reply jobsDB/Footer/Footer.js]: It's not an elegant problem 😂

**Comment by mengtzu** [comment]
NB: I did look at the JobStreet footer and it's not in a great place, we should probably address the debt there soon but not in this PR.

**Pull request merged**